src/main/kotlin/com/exactpro/th2/codec/hand/processor/HandProcessor.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020-2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.codec.hand.processor
+
+import com.exactpro.th2.codec.hand.CodecHandSettings
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import mu.KotlinLogging
+
+/*
+    Following class does RawMessage processing
+    and Message generation
+ */
+abstract class HandProcessor<M, C, R>(
+    private val settings: CodecHandSettings
+) {
+    @Throws(Exception::class)
+    fun process(rawMessage: M, onMessage: (R) -> Unit) {
+        var subSequence = 1
+        val jsonMap: Map<String, Any> = MAPPER.readValue(rawMessage.bodyAsString)
+
+        val context = rawMessage.createContext()
+
+        jsonMap.forEach loop1@ { (key, value) ->
```

Comment:
If you have to use labels maybe it would be better to use `for` loop instead of `forEach` method.
```suggestion
        for ((key, value) in jsonMap) {
```
After that you can also use `continue` to skip the loop iteration